### PR TITLE
store/tikv: handle Canceled error code by grpc remote.

### PR DESCRIPTION
**Pull request by tiancaiamao**
It's observed that killing tikv when running sysbench, we may get a `codes.Canceled` error, that
error is from grpc remote and we didn't handle it properly.

Here I distinguish the error cause of cancel:
1. the request should not retry if cancelled by ourself
2. the request need retry when cancelled by remote 

PTAL @hhkbp2 @disksing @zimulala

### CODE REVIEW

File: store/tikv/region_request_test.go [hunk 1]
```diff
@@ -255,10 +251,20 @@ func (s *testRegionRequestSuite) TestNoReloadRegionForGrpcWhenCtxCanceled(c *C)
...
+	c.Assert(errors.Cause(err), Equals, goctx.Canceled)
 	c.Assert(s.cache.getRegionByIDFromCache(s.region), NotNil)
 
+	// Just for covering error code = codes.Canceled.
+	client1 := &cancelContextClient{
+		Client:       newRPCClient(),
+		redirectAddr: addr,
+	}
+	sender = NewRegionRequestSender(s.cache, client1, kvrpcpb.IsolationLevel_SI)
+	sender.SendReq(s.bo, req, region.Region, 3*time.Second)
```
Comment (disksing) [inline store/tikv/region_request_test.go]: Need some asserts here?

Comment (tiancaiamao) [inline store/tikv/region_request_test.go]: This code is "Just for covering error code = codes.Canceled."
Not testing anything.

**Comment by disksing** [comment]
LGTM.

**Comment by shenli** [comment]
LGTM

**Pull request merged**